components/support/jwcrypto/src/lib.rs

```diff
@@ -0,0 +1,230 @@
+        let protected_header = protected_header
+            .as_ref()
+            .map(|h| serde_json::to_string(&h))
+            .transpose()?
+            .map(|h| base64::encode_config(&h, base64::URL_SAFE_NO_PAD))
+            .unwrap_or_default();
+        let encrypted_key = encrypted_key
+            .as_ref()
+            .map(|k| base64::encode_config(&k, base64::URL_SAFE_NO_PAD))
+            .unwrap_or_default();
+        let iv = iv
+            .as_ref()
+            .map(|iv| base64::encode_config(&iv, base64::URL_SAFE_NO_PAD))
+            .unwrap_or_default();
+        let ciphertext = base64::encode_config(&ciphertext, base64::URL_SAFE_NO_PAD);
+        let auth_tag = auth_tag
+            .as_ref()
+            .map(|t| base64::encode_config(&t, base64::URL_SAFE_NO_PAD))
+            .unwrap_or_default();
+        let jwe_segments = vec![protected_header, encrypted_key, iv, ciphertext, auth_tag];
+        Ok(Self { jwe_segments })
+    }
+
+    fn protected_header(&self) -> Result<Option<JweHeader>> {
+        Ok(self
+            .try_deserialize_base64_segment(0)?
+            .map(|s| serde_json::from_slice(&s))
+            .transpose()?)
+    }
+
+    fn protected_header_raw(&self) -> &str {
+        &self.jwe_segments[0]
+    }
+
+    fn encrypted_key(&self) -> Result<Option<Vec<u8>>> {
+        self.try_deserialize_base64_segment(1)
+    }
+
+    fn iv(&self) -> Result<Option<Vec<u8>>> {
+        self.try_deserialize_base64_segment(2)
+    }
+
+    fn ciphertext(&self) -> Result<Vec<u8>> {
+        Ok(self
+            .try_deserialize_base64_segment(3)?
+            .ok_or_else(|| JwCryptoError::IllegalState("Ciphertext is empty"))?)
+    }
+
+    fn auth_tag(&self) -> Result<Option<Vec<u8>>> {
+        self.try_deserialize_base64_segment(4)
+    }
+
+    fn try_deserialize_base64_segment(&self, index: usize) -> Result<Option<Vec<u8>>> {
+        Ok(match self.jwe_segments[index].is_empty() {
+            true => None,
+            false => Some(base64::decode_config(
+                &self.jwe_segments[index],
+                base64::URL_SAFE_NO_PAD,
+            )?),
+        })
+    }
+}
+
+impl FromStr for CompactJwe {
+    type Err = JwCryptoError;
+    fn from_str(str: &str) -> Result<Self> {
+        let jwe_segments: Vec<String> = str.split('.').map(|s| s.to_owned()).collect();
+        if jwe_segments.len() != 5 {
+            return Err(JwCryptoError::DeserializationError);
+        }
+        Ok(Self { jwe_segments })
+    }
+}
+
+impl ToString for CompactJwe {
+    fn to_string(&self) -> String {
+        assert!(self.jwe_segments.len() == 5);
+        self.jwe_segments.join(".")
+    }
+}
+
+/// Encrypt and serialize data in the JWE compact form.
+pub fn encrypt_to_jwe(data: &[u8], encryption_params: EncryptionParameters) -> Result<String> {
+    let jwe = match encryption_params {
+        EncryptionParameters::ECDH_ES { .. } => ec::encrypt_to_jwe(data, encryption_params)?,
+    };
```

Comment:
Nice cleanup here since we defer all the crypto stuff to the `ec` module. (did I pull too much?)

Reply:
This looks good to me 👍